package.json

```diff
@@ -3,28 +3,38 @@
 	"version": "2.2.0-Preview.1",
 	"description": "Microsoft Graph Client Library",
 	"license": "MIT",
+	"keywords": [
+		"Microsoft",
+		"Graph",
+		"SDK",
+		"JavaScript",
+		"Client"
+	],
 	"main": "lib/src/index.js",
-	"module": "lib/es/index.js",
+	"module": "lib/es/src/index.js",
 	"typings": "lib/src/index",
 	"files": [
 		"lib/",
-		"src/"
+		"src/",
+		"authProviders/"
 	],
 	"types": "./lib/src/index.d.ts",
 	"dependencies": {
 		"@babel/runtime": "^7.4.4",
 		"tslib": "^1.9.3"
 	},
 	"devDependencies": {
+		"@azure/identity": "^1.1.0",
```

Comment:
shouldn't azure identity be in our deps and not dev deps since we have a hard dependency at runtime on it?

Reply:
We are not exporting in the AuthenticationProviders in the main `src/index.ts`.

Reply:
sure but it's in our module resolution path at runtime, and our SDK will fail at runtime if the dependency is not present.

Reply:
Can you give an example of a scenario when you think it will fail?
I created a  local npm package using npm pack script and tried running this example -
// index.js
```
const {Client} = require("@microsoft/microsoft-graph-client");
require("cross-fetch/polyfill");
const client = Client.init({
	defaultVersion: "v1.0",
	debugLogging: true,
	authProvider: (done) => {
		done(null, accessToken);
	},
});
client
	.api("/me")
	.select("displayName")
	.get()
	.then((res) => {
		console.log(res);
	})
	.catch((err) => {
		console.log(err);
	});
```

//package.json dependencies
```
 "dependencies": {
    "@microsoft/microsoft-graph-client": "file:microsoft-microsoft-graph-client-2.1.0-Preview.2.tgz",
    "cross-fetch": "^3.0.6"
  }
```

It worked fine without taking the `azure/identity` as a dependency.
In JS SDK versions < 2.1, we didn't add msal as a dependency and didn't export the ImplicitMSALAuthProviders in the index.ts for the same reason. 

Please let me know if there is something else that can fail this.

Reply:
in your example here, just add 
```JS
const {TokenCredentialAuthenticationProvider} = require("@microsoft/microsoft-graph-client/authProvider/azureTokenCredentials/");
```
at the top.
and assign a new provider to the client, without installing the dep.

Reply:
"peerDependenciesMeta": {
		"msal": {
			"optional": true
		},
		"@azure/identity": {
			"optional": true
		}
	}

This indicates that  '@azure/identity' is an optional peer dependency .
I think this field was lost while merging changes. I readded this.

Reply:
I think this is an improvement as it'll trigger a warning during the install if the current "package" doesn't have it in it's dependency tree. People might still miss the warning though. And people who don't need that dep (because they are not using the auth provider) will get confused because of the warning.

Reply:
The optional field when set true does not trigger a warning if the peer is not present. This caters our requirement. 
If the user doesn't need to use '@azure/identity', he will not be warned. We have an indicator  in the package.json that says we need the optional dependency if the user is using the TokenCredentialAuthProvider. Runtime failure is expected for someone using the TokenCredentialAuthProvider without installing the '@azure/identity' because they shouldn't be using the TokenCredentialAuthProvider class if the TokenCredentials are not used for authentication. 

> If true, the selected peer dependency will be marked as optional by the package manager and the consumer omitting it won't be reported as an error. 
https://yarnpkg.com/configuration/manifest#peerDependenciesMeta.optional

Reply:
and if they need the `TokenCredentialAuthenticationProvider`, they will have to pass in a `TokenCredential` which comes from azure identity forcing them to install the needed dep. Makes sense.
Now last question: do we need msal listed out there as it's a dep from azure identity?

Reply:
If someone uses ImplicitMsalAuthProvider, they will need msal 1.0 . 
msal is not listed as a runtime dependency anymore. It is an optional dependency just like @azure/identity. The exports for the ImplicitMsalAuthProvider have been removed from the main barrel src/index.ts.  The exports were added only in version 2.1 which caused need for adding msal as a runtime dependency.

Reply:
Great. Thanks for walking through the discussion with me!. Resolving